### PR TITLE
[SQL Logic Test] Add support for environment variables

### PR DESCRIPTION
**Pull request by Tishj**
This PR adds support to the unittest for using environment variables in sql

Previously `require-env` required 2 arguments:
`name` and `expected_value`

Now optionally you can leave out the `expected_value`, in which case we only verify that the environment variable is defined.

On top of that, we also now register these environment variables and allow tests to use them in SQL

## Example

```bash
➜  duckdb git:(master) ✗ export env_var_test=1                               
➜  duckdb git:(master) ✗ echo $env_var_test                            
```      

```sql
require-env env_var_test

query I
select ${env_var_test}
----
1
```

Note:
This does not yet work in a result list
```sql
query I
select 1;
----
${env_var_test}
```

Should probably also add this

### CODE REVIEW

**Comment by Tishj** [comment]
Just realized I need to check for name collision with loops/foreach parameters

File: test/sqlite/sqllogic_test_runner.cpp [hunk 1]
```diff
@@ -515,19 +521,35 @@ void SQLLogicTestRunner::ExecuteFile(string script) {
 				}
 			}
 		} else if (token.type == SQLLogicTokenType::SQLLOGIC_REQUIRE_ENV) {
-			if (token.parameters.size() < 2) {
+			if (InLoop()) {
+				parser.Fail("require-env cannot be called in a loop");
+			}
+
+			if (token.parameters.size() != 1 && token.parameters.size() != 2) {
 				parser.Fail("require-env requires 2 arguments: <env name> <env value>");
```
Comment (samansmink) [inline test/sqlite/sqllogic_test_runner.cpp:529]: error message should probably explain both usages with 1 or 2 params

**Review by samansmink — COMMENTED** [review]
added a small comment, also a test would indeed be nice. 

Maybe this is a good opportunity to switch all S3 test cases to use env variables from https://github.com/duckdb/duckdb/pull/5419 instead of the SET variables. Using the new require-env we could instead of checking for S3_TEST_SERVER_AVAILABLE, check for the presence of the AWS credentials in the env. This would also allow easy switching between s3 backends when testing.

**Comment by Tishj** [comment]
I've adjusted all existing [s3] tests::
1. to use require-env to require `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` and `AWS_DEFAULT_REGION` environment variables
2. removed the initial SET statements for these variables
3. replaced the hardcoded later SETs (for resetting the variables) to use the ${env_var} functionality

**Comment by Mytherin** [comment]
Thanks for the PR! LGTM. @samansmink any more thoughts?

File: test/sql/copy/s3/metadata_cache.test [hunk 2]
```diff
@@ -18,7 +27,7 @@ statement ok
 CREATE TABLE test1 as SELECT * FROM range(10,20) tbl(i);
 
 statement ok
-SET s3_secret_access_key='minio_duckdb_user_password';SET s3_access_key_id='minio_duckdb_user';SET s3_region='eu-west-1'; SET s3_endpoint='duckdb-minio.com:9000'; SET s3_use_ssl=false;
+SET s3_endpoint='duckdb-minio.com:9000'; SET s3_use_ssl=false;
```
Comment (samansmink) [inline test/sql/copy/s3/metadata_cache.test:30]: Could this s3 endpoint also be moved into an env variable? then we can use the env vars to easily switch between s3 environments for all tests

**Review by samansmink — COMMENTED** [review]
Thanks for the changes, looks good! I added one small comment

File: test/sql/copy/s3/metadata_cache.test (hunk 2 above)
Comment (Tishj) [reply test/sql/copy/s3/metadata_cache.test:30]: sure, domain + port together or separately?

Comment (samansmink) [reply test/sql/copy/s3/metadata_cache.test:30]: I would say together is probably nicer

Comment (Tishj) [reply test/sql/copy/s3/metadata_cache.test:30]: Now that we're looking at these variables, `SET s3_use_ssl=false;` also comes up a lot
maybe this should be an environment variable?

**Comment by Mytherin** [comment]
Looks good from my side, @samansmink any more comments or is this ready to go?

**Comment by samansmink** [comment]
@Mytherin lgtm!

**Pull request merged**